### ISSUE TITLE
x/sys/unix, windows: needs to track update to syscall

### ISSUE THREAD

**Issue by robpike** [golang/go#10150]
Since 1.4, my hand scan of the changes to the core repo's syscall package yields these CLs, many of which need to be applied to the go.sys repo:

https://go-review.googlesource.com/#/c/1372/
https://go-review.googlesource.com/#/c/1640/
https://go-review.googlesource.com/#/c/1930/
https://go-review.googlesource.com/#/c/1932/
https://go-review.googlesource.com/#/c/1940/
https://go-review.googlesource.com/#/c/2123/
https://go-review.googlesource.com/#/c/2307/
https://go-review.googlesource.com/#/c/2589/
https://go-review.googlesource.com/#/c/2746/
https://go-review.googlesource.com/#/c/2810/
https://go-review.googlesource.com/#/c/2815/
https://go-review.googlesource.com/#/c/2816/
https://go-review.googlesource.com/#/c/2817/
https://go-review.googlesource.com/#/c/3014/
https://go-review.googlesource.com/#/c/3983/
https://go-review.googlesource.com/#/c/4164/
~~https://go-review.googlesource.com/#/c/4330/~~
https://go-review.googlesource.com/#/c/5005/
https://go-review.googlesource.com/#/c/5287/
https://go-review.googlesource.com/#/c/5288/
https://go-review.googlesource.com/#/c/5576/
https://go-review.googlesource.com/#/c/5833/
https://go-review.googlesource.com/#/c/5835/
https://go-review.googlesource.com/#/c/5836/
https://go-review.googlesource.com/#/c/5837/
https://go-review.googlesource.com/#/c/6350/
https://go-review.googlesource.com/#/c/6372/
https://go-review.googlesource.com/#/c/6385/
~~https://go-review.googlesource.com/#/c/6391/~~
https://go-review.googlesource.com/#/c/7123/

This is not a perfect list. Some may be missing and some are internal/syscall and might not apply.

These CLs (and whatever follow, which should be added to this bug) need to be examined and, if appropriated, adapted and applied to go.sys.

More fundamentally, we also need to stop these packages from diverging so much in future. Some process may be required to encourage updating.

From a higher perspective, I am surprised at the amount of churn in a supposedly quiet package.

**Comment by robpike** [comment]
cc @davecheney @enneff

**Comment by robpike** [comment]
cc @davecheney @adg

**Comment by alexbrainman** [comment]
I looked at windows related syscall package changes only. I can see 3.


syscall: use go generate to build zsyscall_windows.go
https://go-review.googlesource.com/#/c/1940/

I have (had for a while ready for review) correspondent go.sys change
windows: use go generate to build zsyscall_windows.go
https://go-review.googlesource.com/#/c/2530/
If someone wants to review it, go ahead.


syscall: Readlink doesn't handle junction on windows
https://go-review.googlesource.com/#/c/2307/

I dropped ball with this change. I will take care of it.


syscall: make mksyscall_windows.go to work even when output does not uses unsafe
https://go-review.googlesource.com/#/c/5005/

This change does not need to be copied to go.sys. Only z*.go go.sys files needs to be re-generated. That has already happened in unrelated go.sys change.


Alex

**Comment by mikioh** [comment]
I would propose to remove networking-related stuff from sys/unix; specifically the following:

1. Subset of routing message parsers for BSD variants
2. Subset of netlink message parsers for Linux
3. BPF (Berkley Packet Filter) for BSD variants
4. LSF (Linux Socket Filter) for Linux

Simply net sub repository seems to be right place for them. Thoughts?

**Comment by robpike** [comment]
Mikioh: I don't know enough about networking to guarantee whether this is OK, but that sounds reasonable to me.

Alex: 2530 LGTM. Thanks.

**Comment by robpike** [comment]
https://go-review.googlesource.com/#/c/6555/

**Comment by robpike** [comment]
https://go-review.googlesource.com/#/c/4913/ also needs to be applied

**Comment by robpike** [comment]
https://go-review.googlesource.com/7543 also needs to be applied

**Comment by 4ad** [comment]
https://go-review.googlesource.com/8022/ also needs to be applied.

**Comment by robpike** [comment]
https://go-review.googlesource.com/8022 has been applied to x/sys by https://go-review.googlesource.com/8023

**Comment by robpike** [comment]
https://go-review.googlesource.com/7940
https://go-review.googlesource.com/6701

**Comment by robpike** [comment]
https://go-review.googlesource.com/#/c/2167

**Comment by robpike** [comment]
https://go-review.googlesource.com/#/c/8721/ handles 6350 and 6385

**Comment by robpike** [comment]
https://go-review.googlesource.com/#/c/8720/ handles 2167

**Comment by robpike** [comment]
http://golang.org/cl/5833
http://golang.org/cl/5835
http://golang.org/cl/5837

are done by https://go-review.googlesource.com/10032

**Comment by bradfitz** [comment]
This style of bug is hard to follow or close.

N smaller, closable bugs might be better.

**Comment by robpike** [comment]
I like having it in one.

http://golang.org/cl/8723 
fixed by
https://go-review.googlesource.com/10035

**Comment by 4ad** [comment]
Perhaps a milestone? Whatever github has to group issues.

**Comment by robpike** [comment]
https://golang.org/cl/9870/
fixed by
https://go-review.googlesource.com/#/c/10036/

I will create a new bug with only the remaining ones, since people seem offended by _this_ particular noise source when there is so much other noise out there.
